postgres/datadog_checks/postgres/config.py

```diff
@@ -167,6 +168,35 @@ def _build_tags(self, custom_tags):
         if rds_tags:
             tags.extend(rds_tags)
         return tags
+    
+    def _get_max_connections_per_thread(self):
+        """
+        Returns the maximum number of connections per thread.
+        i.e.
+        max_connections = 30
+        When dbm_enabled = True, we have 4 threads:
+            - 1 thread for the main integration
+            - 1 thread for statement_metrics
+            - 1 thread for metadata_samples
```

Comment:
if schema collection is enabled, we need to take into account that the metadata thread can open many connections as well